core/src/main/java/org/axonframework/config/SagaConfiguration.java

```diff
@@ -33,11 +36,14 @@
     }
 
     private SagaConfiguration(Class<S> sagaType) {
+        String resourceInjectorName = sagaType.getSimpleName() + "ResourceInjector";
         String managerName = sagaType.getSimpleName() + "Manager";
         String processorName = sagaType.getSimpleName() + "Processor";
         String repositoryName = sagaType.getSimpleName() + "Repository";
+        resourceInjector = new Component<>(() -> config, resourceInjectorName, c -> new NoResourceInjector());
```

Comment:
I think the default here should take the resourceInjector from the global configuration. That would allow a single config for all places where the resourceInjector is being used, while it's still possible to configure a separate one per Saga.